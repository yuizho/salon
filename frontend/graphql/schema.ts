/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.

export type AddOperation = {
  __typename: "AddOperation",
  room_id: string,
  user_id: string,
  user_token: string,
};

export type UpdateOperation = {
  __typename: "UpdateOperation",
  room_id: string,
  user_id: string,
};

export enum Status {
  CHOOSING = "CHOOSING",
  CHOSEN = "CHOSEN",
  LEAVED = "LEAVED",
}


export type User = {
  __typename: "User",
  room_id: string,
  user_id: string,
  status: Status,
  picked_card?: string | null,
  operated_at: string,
};

export type Room = {
  __typename: "Room",
  items:  Array<User >,
  expiration_unix_timestamp: number,
  is_opened: boolean,
};

export type OpenRoomMutation = {
  openRoom:  {
    __typename: "AddOperation",
    room_id: string,
    user_id: string,
    user_token: string,
  },
};

export type JoinMutationVariables = {
  room_id: string,
};

export type JoinMutation = {
  join:  {
    __typename: "AddOperation",
    room_id: string,
    user_id: string,
    user_token: string,
  },
};

export type LeaveMutationVariables = {
  room_id: string,
  user_id: string,
  user_token: string,
};

export type LeaveMutation = {
  leave:  {
    __typename: "UpdateOperation",
    room_id: string,
    user_id: string,
  },
};

export type PickMutationVariables = {
  room_id: string,
  user_id: string,
  picked_card: string,
  user_token: string,
};

export type PickMutation = {
  pick:  {
    __typename: "UpdateOperation",
    room_id: string,
    user_id: string,
  },
};

export type RefreshTableMutationVariables = {
  room_id: string,
  user_id: string,
  user_token: string,
};

export type RefreshTableMutation = {
  refreshTable:  {
    __typename: "UpdateOperation",
    room_id: string,
    user_id: string,
  },
};

export type KickMutationVariables = {
  room_id: string,
  user_id: string,
  kicked_user_id: string,
  user_token: string,
};

export type KickMutation = {
  kick:  {
    __typename: "UpdateOperation",
    room_id: string,
    user_id: string,
  },
};

export type UpdateUserMutationVariables = {
  room_id: string,
  user_id: string,
  status: Status,
  picked_card?: string | null,
  operated_at: string,
};

export type UpdateUserMutation = {
  updateUser:  {
    __typename: "User",
    room_id: string,
    user_id: string,
    status: Status,
    picked_card?: string | null,
    operated_at: string,
  },
};

export type GetRoomQueryVariables = {
  room_id: string,
};

export type GetRoomQuery = {
  getRoom:  {
    __typename: "Room",
    items:  Array< {
      __typename: "User",
      room_id: string,
      user_id: string,
      status: Status,
      picked_card?: string | null,
      operated_at: string,
    } >,
    expiration_unix_timestamp: number,
    is_opened: boolean,
  },
};

export type OnUpdateUserSubscriptionVariables = {
  room_id: string,
};

export type OnUpdateUserSubscription = {
  onUpdateUser?:  {
    __typename: "User",
    room_id: string,
    user_id: string,
    status: Status,
    picked_card?: string | null,
    operated_at: string,
  } | null,
};
