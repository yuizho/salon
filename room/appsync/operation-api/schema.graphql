type Query {
  getOperation(event_id: String!): Operation
}

type Mutation {
  openRoom: Operation
  join(room_id: String!): Operation
  leave(room_id: String!, user_id: String!): Operation
  pick(room_id: String!, user_id: String!, picked_card: String!): Operation
  refreshTable(room_id: String!, user_id: String!): Operation
}

type Operation {
  event_id: String
  room_id: String
  user_id: String
  op_type: String
  operated_at: String
  picked_card: String
}
