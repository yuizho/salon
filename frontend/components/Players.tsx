import { FC } from 'react';

import Player from './Player';

type Props = {
  myUserId: string;
  players: Array<{
    userId: string;
    pickedCard: string;
  }>;
  shouldOpenCards: boolean;
};

const Players: FC<Props> = ({ myUserId, players, shouldOpenCards }) => (
  <div
    className={`
  flex flex-wrap gap-2
  `}
  >
    {players.map((u) => {
      const isMe = u.userId === myUserId;
      const isShown = shouldOpenCards || isMe;
      return (
        <Player
          key={u.userId}
          userId={u.userId}
          value={u.pickedCard}
          isShown={isShown}
          isMe={isMe}
        />
      );
    })}
  </div>
);

export default Players;
