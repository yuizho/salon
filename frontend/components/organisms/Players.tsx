import { FC } from 'react';

import Player from './Player';

type Props = {
  myUserId: string;
  players: Array<{
    userId: string;
    pickedCard: string;
  }>;
  shown: boolean;
};

const Players: FC<Props> = ({ myUserId, players, shown }) => (
  <div
    className={`
  flex flex-wrap gap-2
  `}
  >
    {players.map((u) => {
      const me = u.userId === myUserId;
      const showThisCard = shown || me;
      return (
        <Player
          key={u.userId}
          userId={u.userId}
          value={u.pickedCard}
          shown={showThisCard}
          me={me}
        />
      );
    })}
  </div>
);

export default Players;
