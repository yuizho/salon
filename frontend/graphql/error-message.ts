// eslint-disable-next-line import/prefer-default-export
export const NETWORK_ERROR =
  '通信エラーが発生しました。ネットワークの状態を確認し、解決しない場合は時間をおいてからやり直してください。';
