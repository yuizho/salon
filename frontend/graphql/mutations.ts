/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const openRoom = /* GraphQL */ `
  mutation OpenRoom {
    openRoom {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const join = /* GraphQL */ `
  mutation Join($room_id: String!) {
    join(room_id: $room_id) {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const leave = /* GraphQL */ `
  mutation Leave($room_id: String!, $user_id: String!) {
    leave(room_id: $room_id, user_id: $user_id) {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const pick = /* GraphQL */ `
  mutation Pick($room_id: String!, $user_id: String!, $picked_card: String!) {
    pick(room_id: $room_id, user_id: $user_id, picked_card: $picked_card) {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const refreshTable = /* GraphQL */ `
  mutation RefreshTable($room_id: String!, $user_id: String!) {
    refreshTable(room_id: $room_id, user_id: $user_id) {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const kick = /* GraphQL */ `
  mutation Kick(
    $room_id: String!
    $user_id: String!
    $kicked_user_id: String!
  ) {
    kick(
      room_id: $room_id
      user_id: $user_id
      kicked_user_id: $kicked_user_id
    ) {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const updateUser = /* GraphQL */ `
  mutation UpdateUser(
    $room_id: String!
    $user_id: String!
    $status: Status!
    $picked_card: String
    $operated_at: AWSDateTime!
  ) {
    updateUser(
      room_id: $room_id
      user_id: $user_id
      status: $status
      picked_card: $picked_card
      operated_at: $operated_at
    ) {
      room_id
      user_id
      status
      picked_card
      operated_at
    }
  }
`;
