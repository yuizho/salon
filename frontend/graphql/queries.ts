/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const getRoom = /* GraphQL */ `
  query GetRoom($room_id: String!) {
    getRoom(room_id: $room_id) {
      items {
        room_id
        user_id
        status
        picked_card
        operated_at
      }
      expiration_unix_timestamp
      is_opened
    }
  }
`;
