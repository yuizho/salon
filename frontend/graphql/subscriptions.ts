/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const onUpdateUser = /* GraphQL */ `
  subscription OnUpdateUser($room_id: String!) {
    onUpdateUser(room_id: $room_id) {
      room_id
      user_id
      status
      picked_card
      operated_at
    }
  }
`;
