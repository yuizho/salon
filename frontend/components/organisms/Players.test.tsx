import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import { Status } from '../../graphql/schema';

import Players from './Players';

describe('Players', () => {
  test('snapshot', () => {
    const { asFragment } = render(
      <RecoilRoot>
        <Players
          myUserId="xxx"
          players={[
            { userId: 'xxx', status: Status.CHOSEN, pickedCard: 'XL' },
            { userId: 'yyy', status: Status.CHOSEN, pickedCard: 'S' },
            { userId: 'zzz', status: Status.CHOSEN, pickedCard: 'M' },
          ]}
          shown={true}
        />
      </RecoilRoot>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
