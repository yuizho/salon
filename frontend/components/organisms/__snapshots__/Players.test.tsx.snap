// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players snapshot 1`] = `
<DocumentFragment>
  <div
    class="
  flex flex-wrap gap-2
  "
  >
    <div
      class="flex flex-col items-center gap-2 p-2 w-20"
    >
      <div
        class="
      flex
      flex-col
      items-center
      w-20
    "
      >
        <div
          class="
        flex
        items-center justify-center
        h-14
        w-14
        rounded-full
        border
        bg-white
        border-slate-400
      "
        >
          <div
            class="relative text-slate-600"
          >
            <div
              class="text-3xl"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M4 22a8 8 0 1 1 16 0H4zm8-9c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="relative"
      >
        <div
          class="relative"
        >
          <div
            class="
      
flex
items-center justify-center
w-14 h-20
border-slate-400
rounded border
text-slate-600
select-none
bg-white

      
      
      font-bold text-2xl
    "
          >
            XL
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col items-center gap-2 p-2 w-20"
    >
      <div
        class="
      flex
      flex-col
      items-center
      w-20
    "
      >
        <div
          class="
        flex
        items-center justify-center
        h-14
        w-14
        rounded-full
        border
        bg-white
        
border-slate-400
shadow-md
hover:cursor-pointer
hover:border-2
active:animate-none
active:shadow-none
active:border
active:border-slate-300

      "
        >
          <div
            class="relative text-slate-600"
          >
            <div
              class="text-3xl"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M4 22a8 8 0 1 1 16 0h-2a6 6 0 1 0-12 0H4zm8-9c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm0-2c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="relative"
      >
        <div
          class="relative"
        >
          <div
            class="
      
flex
items-center justify-center
w-14 h-20
border-slate-400
rounded border
text-slate-600
select-none
bg-white

      
      
      font-bold text-2xl
    "
          >
            S
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col items-center gap-2 p-2 w-20"
    >
      <div
        class="
      flex
      flex-col
      items-center
      w-20
    "
      >
        <div
          class="
        flex
        items-center justify-center
        h-14
        w-14
        rounded-full
        border
        bg-white
        
border-slate-400
shadow-md
hover:cursor-pointer
hover:border-2
active:animate-none
active:shadow-none
active:border
active:border-slate-300

      "
        >
          <div
            class="relative text-slate-600"
          >
            <div
              class="text-3xl"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M4 22a8 8 0 1 1 16 0h-2a6 6 0 1 0-12 0H4zm8-9c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm0-2c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="relative"
      >
        <div
          class="relative"
        >
          <div
            class="
      
flex
items-center justify-center
w-14 h-20
border-slate-400
rounded border
text-slate-600
select-none
bg-white

      
      
      font-bold text-2xl
    "
          >
            M
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
